cpp/src/parquet/column_writer.cc

```diff
@@ -1704,10 +1779,15 @@ Status TypedColumnWriterImpl<DType>::WriteArrowDictionary(
       referenced_dictionary = referenced_dictionary_datum.make_array();
     }
 
-    int64_t non_null_count = chunk_indices->length() - chunk_indices->null_count();
-    page_statistics_->IncrementNullCount(num_chunk_levels - non_null_count);
-    page_statistics_->IncrementNumValues(non_null_count);
-    page_statistics_->Update(*referenced_dictionary, /*update_counts=*/false);
+    if (page_statistics_) {
+      int64_t non_null_count = chunk_indices->length() - chunk_indices->null_count();
+      page_statistics_->IncrementNullCount(num_chunk_levels - non_null_count);
+      page_statistics_->IncrementNumValues(non_null_count);
+      page_statistics_->Update(*referenced_dictionary, /*update_counts=*/false);
+    }
+    if (page_size_stats_builder_) {
+      page_size_stats_builder_->WriteValues(*referenced_dictionary);
```

Comment:
is this write, why are we writing values in the dictionary for page size stats?  Maybe a comment or a a name value?

Reply:
Yes, this is write. The binary values are passed in the form of an `arrow::DictionaryArray` and encoded indices in an `arrow::Int32Array`. Here we need to restore the referenced values in the dictionary array to precisely build page stats and size stats.